.ci/release/Jenkinsfile

```
@@ -0,0 +1,238 @@
+
+@Library('apm@current') _
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-agent-java'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    DOCKERHUB_SECRET = 'secret/apm-team/ci/elastic-observability-dockerhub'
+    ELASTIC_DOCKER_SECRET = 'secret/apm-team/ci/docker-registry/prod'
+    NEXUS_SECRET = 'secret/apm-team/ci/nexus'
+    MAVEN_CONFIG = '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dmaven.repo.local=.m2'
+    OPBEANS_REPO = 'opbeans-java'
+
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+
+  }
+  parameters {
+    string(name: 'branch_specifier', defaultValue: 'master')
+  }
+
+  stages {
+    stage('Initializing'){
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        JAVA_HOME = "${env.HUDSON_HOME}/.java/java10"
+        PATH = "${env.JAVA_HOME}/bin:${env.PATH}"
+      }
+      stages {
+        /**
+         Checkout the code
+        */
+        stage('Checkout') {
+          steps {
+            dir("${BASE_DIR}") {
+              // This is required to ensure that Maven is working with a real branch
+              sh(script: "git checkout -f {$branch_specifier}")
+            }
+          }
+        }
+      }
+    }
+    stage('Release') {
+      options { skipDefaultCheckout () }
+      stages{
+        // 1. Check to see if oss.sonatype.org is up. We throw an exception and fail the job if there is no response.
+        stage('Check oss.sonatype.org') {
+          steps {
+            // If this fails, an exception should be thrown and execution will halt
+            httpRequest(url: "https://oss.sonatype.org/#welcome")
+          }
+        }
+        // 1.1 Check to ensure the build is green. Require confirmation if it is not.
+        stage('Check master build status') {
+          steps {
+            script {
+              // If this build is not green: https://apm-ci.elastic.co/job/apm-agent-java/job/apm-agent-java-mbp/job/master/
+              if(!buildStatus(host: 'apm-ci.elastic.co', job: ['apm-agent-java', 'apm-agent-java-mbp', 'master'], return_boolean: true)) {
+                input(message: "WARNING! The master build is not passing. Do you wish to continue?")
+              }
+            }
+          }
+        }
+        // 2. Review project version.
+        stage('Review project version') {
+          steps {
+            // 2.1 Fetch the current version from pom.xml
+            def ver = mvnVersion(showQualifiers: true)
+            // 2.2 Raise a prompt asking if the version is correct or if they wish to update
+            script {
+              def should_continue = input(message: "Current version is ${ver}", parameters: [
```

Moved to the script closure

```suggestion
            script {
              // 2.1 Fetch the current version from pom.xml
              def ver = mvnVersion(showQualifiers: true)
              // 2.2 Raise a prompt asking if the version is correct or if they wish to update
              def should_continue = input(message: "Current version is ${ver}", parameters: [
```